lib/tests/streamlit/caching_test.py

```diff
@@ -39,6 +39,55 @@ def foo():
         self.assertEqual(foo(), 42)
         self.assertEqual(foo(), 42)
 
+    def test_ignore_changes_to_global_constants(self):
+        # Changing values of closure variables should not change cache value after first cache.
+        # See PR #817 / Issue #789
+        number = 5
+
+        @st.cache
+        def score():
+            return number + 2
+            result = {
```

Comment:
`def score()` needs cleanup, you're returning twice and there is no `return_things()` defined

Reply:
weird.  removed.